### DISCUSSION TITLE
Event Schemas with Pydantic

### DISCUSSION THREAD

**Discussion by d-chambers** [niosh-mining/obsplus#216] [Ideas]
I am looking into tearing out ObsPlus' hinky code for json conversion and replacing it with [Pydantic](https://pydantic-docs.helpmanual.io/) models in the [event_schema branch](https://github.com/niosh-mining/obsplus/tree/event_schema). It looks promising so far. 

The only downside is that we will need to take pydantic as a hard dependency. I am not worried about the reliability of the project as it is well supported and used by several mainstream packages (fast-api, typer). Nothing we are doing at NIOSH would have a problem with taking it as a new dependency but I want to check with other users. Particularly @calum-chamberlain for [RT_EQCorrScan ](https://github.com/eqcorrscan/RT_EQcorrscan).

**Comment by calum-chamberlain** [comment]
I'm pretty sure I don't have a problem with adding pydantic as a dependency - RTEQcorrscan already has a hideous list of dependencies, as long as pydantic doesn't clash with anything else (it has pretty minimal dependencies itself, and I don't think any should be a problem).

Would this replace `cat_to_dict` and `dict_to_cat`?

**Comment by d-chambers** [comment]
Ok sounds good.

This wouldn't replace those functions, just provide a more straight forward way of doing the conversion. The ObsPlus public API shouldn't change much, if at all.
